src/views/Nft/market/hooks/useCollectionNfts.tsx

```diff
@@ -0,0 +1,161 @@
+import { useEffect, useState } from 'react'
+import { ApiResponseCollectionTokens, NftAttribute, NftToken } from 'state/nftMarket/types'
+import { useGetNftFilters, useGetNftOrdering, useGetNftShowOnlyOnSale } from 'state/nftMarket/hooks'
+import {
+  fetchNftsFiltered,
+  getMarketDataForTokenIds,
+  getNftApi,
+  getNftsFromCollectionApi,
+  getNftsMarketData,
+} from 'state/nftMarket/helpers'
+// eslint-disable-next-line camelcase
+import useSWRInfinite from 'swr/infinite'
+import isEmpty from 'lodash/isEmpty'
+import uniqBy from 'lodash/uniqBy'
+import { REQUEST_SIZE } from '../Collection/config'
+
+interface ItemListingSettings {
+  field: string
+  direction: 'asc' | 'desc'
+  showOnlyNftsOnSale: boolean
+  nftFilters: Record<string, NftAttribute>
+}
+
+const fetchTokenIdsFromFilter = async (address: string, settings: ItemListingSettings) => {
+  const filterObject: Record<string, NftAttribute> = settings.nftFilters
+  const attrParams = Object.values(filterObject).reduce(
+    (accum, attr) => ({
+      ...accum,
+      [attr.traitType]: attr.value,
+    }),
+    {},
+  )
+  const attrFilters = !isEmpty(attrParams) ? await fetchNftsFiltered(address, attrParams) : null
+  return attrFilters ? Object.values(attrFilters.data).map((apiToken) => apiToken.tokenId) : null
+}
+
+const fetchMarketDataNfts = async (address: string, settings: ItemListingSettings, page: number) => {
```

Comment:
define return type for `fetchMarketDataNfts` and `fetchAllNfts` for consistency.

Reply:
Done